### PR TITLE
BUGFIX: use Guzzle to download from GitHub

### PR DESCRIPTION
**Pull request by Pingu501**
Previously used `file_get_contents ` does ignore all PHP proxy settings. The GuzzleHTTPClient resolves this problem

### CODE REVIEW

File: Classes/Composer/InstallerScripts.php [hunk 1]
```diff
@@ -71,8 +72,10 @@ public static function postUpdateAndInstall(Event $event)
             echo '> Version:      ' . $version . $versionMessage . "\n";
             echo '> Platform:     ' . $platform . "\n";
             echo '> Architecture: ' . $architecture . "\n";
+
+            $httpClient = new Client();
             $downloadLink = sprintf('https://github.com/Flowpack/prunner/releases/download/v%s/prunner_%s_%s_%s.tar.gz', $version, $version, $platform, $architecture);
-            $downloadedFileContents = file_get_contents($downloadLink);
+            $downloadedFileContents = $httpClient->get($downloadLink)->getBody()->getContents();
```
Comment (hlubek) [inline Classes/Composer/InstallerScripts.php]: We could optimize this if we directly open a stream to the target file (`GuzzleHttp\Psr7\Utils::streamFor($resource)`) and call Guzzle via `$client->request('GET', $downloadLink, ['save_to' => $stream]);`

Comment (Pingu501) [reply Classes/Composer/InstallerScripts.php]: now using the `sink` property to download directly to file

File: Classes/Composer/InstallerScripts.php [hunk 2]
```diff
@@ -71,8 +72,10 @@ public static function postUpdateAndInstall(Event $event)
...
             echo '> Platform:     ' . $platform . "\n";
             echo '> Architecture: ' . $architecture . "\n";
+
             $downloadLink = sprintf('https://github.com/Flowpack/prunner/releases/download/v%s/prunner_%s_%s_%s.tar.gz', $version, $version, $platform, $architecture);
-            $downloadedFileContents = file_get_contents($downloadLink);
+            $httpClient = new Client();
+            $httpClient->get($downloadLink, ['sink' => 'Data/Temporary/prunner.tar.gz']);
             echo '> Download complete.' . "\n";
 
             file_put_contents('Data/Temporary/prunner.tar.gz', $downloadedFileContents);
```
Comment (hlubek) [inline Classes/Composer/InstallerScripts.php:81]: I think this has to be removed then ;)

File: Classes/Composer/InstallerScripts.php [hunk 3]
```diff
@@ -71,8 +72,10 @@ public static function postUpdateAndInstall(Event $event)
             echo '> Version:      ' . $version . $versionMessage . "\n";
             echo '> Platform:     ' . $platform . "\n";
             echo '> Architecture: ' . $architecture . "\n";
+
+            $httpClient = new Client();
             $downloadLink = sprintf('https://github.com/Flowpack/prunner/releases/download/v%s/prunner_%s_%s_%s.tar.gz', $version, $version, $platform, $architecture);
```
Comment (Benjamin-K) [inline Classes/Composer/InstallerScripts.php:76]: I'd replace this with
```php
$downloadLink = sprintf('https://github.com/Flowpack/prunner/releases/download/v%1$s/prunner_%1$s_%2$s_%3$s.tar.gz', $version, $platform, $architecture);
```
so you do not have to write the version twice.

File: Classes/Composer/InstallerScripts.php (hunk 2 above)
Comment (Benjamin-K) [reply Classes/Composer/InstallerScripts.php:81]: CC @Pingu501

**Review by Benjamin-K — CHANGES_REQUESTED** [review]
Currently the file will probably be overridden. So `file_put_contents` needs to be removed.

**Comment by Benjamin-K** [comment]
@hlubek The `file_put_contents` line needs to be removed, or not?

**Comment by hlubek** [comment]
Released with 0.3.0

**Pull request merged**